base/math.jl

```diff
@@ -758,17 +758,34 @@ end
 atan(y::Real, x::Real) = atan(promote(float(y),float(x))...)
 atan(y::T, x::T) where {T<:AbstractFloat} = Base.no_op_err("atan", T)
 
-max(x::T, y::T) where {T<:AbstractFloat} = ifelse((y > x) | (signbit(y) < signbit(x)),
-                                    ifelse(isnan(x), x, y), ifelse(isnan(y), y, x))
-
-
-min(x::T, y::T) where {T<:AbstractFloat} = ifelse((y < x) | (signbit(y) > signbit(x)),
-                                    ifelse(isnan(x), x, y), ifelse(isnan(y), y, x))
+_isless(x::T, y::T) where {T<:AbstractFloat} = (x < y) || (signbit(x) > signbit(y))
+min(x::T, y::T) where {T<:AbstractFloat} = isnan(x) || ~isnan(y) && _isless(x, y) ? x : y
+max(x::T, y::T) where {T<:AbstractFloat} = isnan(x) || ~isnan(y) && _isless(y, x) ? x : y
+minmax(x::T, y::T) where {T<:AbstractFloat} = min(x, y), max(x, y)
+
+_isless(x::Float16, y::Float16) = signbit(widen(x) - widen(y))
+
+function min(x::T, y::T) where {T<:Union{Float32,Float64}}
```

Comment:
Should the new `min`, `max`, `minmax` signatures be expanded to include `Float16`?

Reply:
using `T<:IEEEFloat` everywhere that now uses 
`T<:Union{Float32,Float64}` is helpful.  Some systems have onchip support for Float16s, so this is a win there.  The systems that emulate Float16s support min, max, minmax, so the processing is at worst unchanged there.